src/Resources/assets/shop/js/elasticSearchAutocomplete.js

```diff
@@ -0,0 +1,133 @@
+        this.defaultConfig = {
+            searchFields: '.searchdiv',
+            baseAutocompleteVariantUrl: '[data-bb-elastic-url]',
+            searchInput: '.app-quick-add-code-input',
+            resultsTarget: '.results',
+            resultContainerClassesArray: ['result'],
+            resultImageClass: 'image',
+            resultContentClass: 'result__content',
+            resultPriceClass: 'result__price',
+            resultTitleClass: 'js-title',
+            resultDescriptionClass: 'result__description',
+            resultLinkClass: 'result__link',
+            resultCategoryClass: 'result__category',
+            resultImageClass: 'result__image',
+            resultContainerClass: 'result__container',
+        };
+        this.finalConfig = {...this.defaultConfig, ...config};
+        this.searchFieldsSelector = document.querySelector(this.finalConfig.searchFields);
+    }
+
+    init() {
+        if (this.config && typeof this.config !== 'object') {
+            throw new Error('BitBag - CreateConfirmationModal - given config is not valid - expected object');
+        }
+
+        this._debounce();        
+    }
+
+    _toggleModalVisibility(elements) {
+        document.addEventListener('variantsVisible', () => {
+            document.addEventListener('click', () => {
+                elements.forEach((element) => {
+                    element.innerHTML = '';
+                    element.style.display = 'none';
+                });
+            });
+        });
+    }
+
+    _modalTemplate(item, categoryStyle) {
+        const result = document.createElement('a');
+        result.classList.add(...this.finalConfig.resultContainerClassesArray, 'js-result');
+        result.innerHTML = `
+            <h3 class=${this.finalConfig.resultCategoryClass} style=${categoryStyle}>${item.taxon_name}</h3> 
+                <a href=${item.slug} class=${this.finalConfig.resultLinkClass}>
+                    <div class=${this.finalConfig.resultContainerClass}>
+                        <img class=${this.finalConfig.resultImageClass} src=${item.image}>
+                        <div class=${this.finalConfig.resultContentClass}>
+                            <div class=${this.finalConfig.resultTitleClass}>${item.name}</div>
+                            <div class=${this.finalConfig.resultPriceClass}>${item.price}</div>
+                        </div>
+                    </div>
+                </a> 
+        `;
+
+        return result;
+    }
+
+    _assignElements(entry, data) {
+        const currentResults = this.searchFieldsSelector.querySelector(this.finalConfig.resultsTarget);
+       
+        currentResults.innerHTML = ''
+        currentResults.style = 'visibility: visible';
+
+        const allResults = document.querySelectorAll(this.finalConfig.resultsTarget);
+         
+        if (data.items.length === 0) {
+            currentResults.innerHTML = '<center class="result">no matching results</center>';
+        }
```

Comment:
is it needed? backend wont throw error when there are no results? and it is not handled in catch(error) block?

Reply:
im not sure about it but its worth to check that